src/bip47-test.cpp

```diff
@@ -0,0 +1,273 @@
+    purposeKey.Derive(coinTypeKey, 0);
+    coinTypeKey.Derive(identityKey, 0);
+
+
+
+//    CExtKey key;
+    unsigned char data[80];
+    CExtPubKey pubkey;
+    pubkey = identityKey.Neuter();
+    identityKey.Encode(data);
+    pubkey.Encode(data);
+
+    bip47::byte alicepubkey[33];
+    bip47::byte alicechian[32];
+    std::copy(pubkey.pubkey.begin(), pubkey.pubkey.end(), alicepubkey);
+    std::copy(pubkey.chaincode.begin(), pubkey.chaincode.end(), alicechian);
+
+    bip47::PaymentCode alicePcode(alicepubkey, alicechian);
+    printf("\n Payment code of alice \n%s\n", alicePcode.ToString().c_str());
+
+
+    printf("Encoded Data\n");
+    for(int i = 0; i < 80; i++) {
+        printf("%u", data[i]);
+    }
+    printf("\n");
+
+    CBitcoinExtKey b58key; b58key.SetKey(identityKey);
+
+    printf("%s\n", b58key.ToString().c_str());
+
+    CBitcoinExtPubKey b58pubkey; b58pubkey.SetKey(pubkey);
+
+    printf("Pubkey value is %s\n",b58pubkey.ToString().c_str());
+
+    std::string strPcode = "PM8TJTLJbPRGxSbc8EJi42Wrr6QbNSaSSVJ5Y3E4pbCYiTHUskHg13935Ubb7q8tx9GVbh2UuRnBc3WSyJHhUrw8KhprKnn9eDznYGieTzFcwQRya4GA";
+    bip47::PaymentCode paymentCode(strPcode);
+
+    CPubKey masterPubkey = paymentCode.getPubkey();
+    printf("\n master pubkey size %d\n", masterPubkey.size());
+
+    if (masterPubkey.IsValid()) {
+        printf("\nmaster Pubkey is valid\n");
+        CBitcoinAddress address(masterPubkey.GetID());
+        std::cout << " Address from masterPubkey " << address.ToString() << std::endl;
+    } else {
+        printf("\nmaster Pubkey is not valid\n");
+    }
+
+
+    CExtPubKey extPubKey;
+    paymentCode.valid();
+    bip47::byte payloads[80] = {};
+    paymentCode.get_payload(payloads);
+
+
+    CBitcoinAddress newaddress("TN5pADYZMSDyKpcgicC1d2Z2adHdTgHRGG");
+    CKeyID keyID;
+    newaddress.GetKeyID(keyID);
+    CPubKey vchPubKey;
+
+
+//    "File named wallet.dat"
+//    if (!pwalletMain->GetPubKey(keyID, vchPubKey)) {
+//        printf("\n findout pubkey of address\n");
+//    } else {
+//        printf("\n cannot find out pubkey of address\n");
+//    }
+
+//    bip47::PaymentCode paymentCode1()
+
+
+
+    /**
+     * CKey CPubKey
+     * CKeyMetadata
+     * CBitcoinExtKey
+     * CBitcoinExtPubKey
+     * CExtKey
+     * CExtPubKey
+     */
+
+//    SelectParams(CBaseChainParams::REGTEST);
+
+//    string strAddress = "TPj1wZxMM7TRWeBdKWWMn34G6XNeobCq9K";
+//    string privstr = "cSaagnTkEJymA6amdQ5kpdPTszfKzdjiXZmJm42qy7Fd4MnTwZeB";
+//    CBitcoinSecret vchSecret;
+//    vchSecret.SetString(privstr);
+//    CKey key = vchSecret.GetKey();
+//    printf("%s\n", key.IsValid() ? "true" : "false");
+//
+//    LOCK(pwalletMain->cs_wallet);
+//    CPubKey pubKey = key.GetPubKey();
+
+
+//    assert(key.VerifyPubKey(pubKey));
+
+
+//    CBitcoinAddress addr(strAddress);
+//    CKeyID keyId;
+//    addr.GetKeyID(keyId);
+
+//    pwalletMain->GetKey(hdChain.masterKeyID, key);
```

Comment:
Same question about this commented out code

Reply:
Removed